app/src/main/java/com/example/next_show/data/ShowFilter.java

```diff
@@ -152,26 +146,19 @@ private boolean matchedNetworkFilter(Show current, Set<String> networks) {
         return false;
     }
 
-    public static List<Show> getGenreMatch(List<Show> shows, Set<String> compareGenres){
-        // init return list
-        List<Show> genreMatched = new ArrayList<>();
-
-        // check if we need to even compare
+    public boolean matchedGenreFilter(Show current, Set<String> compareGenres) {
         if (compareGenres.isEmpty()) {
-            return genreMatched;
+            return true;
         }
 
-        for (Show s: shows) {
-            Set<String> showGenres = new HashSet<String>(s.getGenres());
-            showGenres.addAll(s.getGenres());
+        Set<String> showGenres = new HashSet<String>();
+        showGenres.addAll(current.getGenres());
+        showGenres.retainAll(compareGenres);
 
-            // if any genre(s) match the comparison ones or if set intersection is not empty
-            showGenres.retainAll(compareGenres);
-            if (!showGenres.isEmpty()) {
-                genreMatched.add(s);
-            }
+        // if genres match
+        if (!showGenres.isEmpty()) {
+            return true;
         }
-
-        return genreMatched;
+        return false;
```

Comment:
You can condense this to just:

```
return !showGenres.isEmpty()
```

Generally if you're just returning a boolean in an if/else block without doing any other work, you can just return the outer condition that's being used to determine if it goes into the true or false block